docs/import/color_function_util.js

```diff
@@ -266,49 +253,96 @@ function withColor(td, layer, property) {
   if (!colorFunction) {
     return td.text('N/A').addClass('na');
   }
-  const {currentStyle} = colorFunction;
-  switch (currentStyle) {
+  switch (colorFunction.currentStyle) {
     case ColorStyle.SINGLE:
-      td.append(createColorBox(colorFunction['color']));
+      td.append(createColorBox(colorFunction.color));
       break;
     case ColorStyle.CONTINUOUS:
-      td.append(createColorBox(colorFunction['color']));
+      td.append(createColorBox(colorFunction.color));
       break;
     case ColorStyle.DISCRETE:
       createColorBoxesForDiscrete(colorFunction, td);
       break;
     default:
       setStatus(ILLEGAL_STATE_ERR + 'unrecognized color function: ' + layer);
   }
-  td.addClass('editable color-td').on('click', () => onClick(td, currentStyle));
+  td.addClass('editable color-td').on('click', () => onClick(td));
   return td;
 }
 
 /**
  * On click function for color tds - updates the globalTd.
  * @param {JQuery<HTMLElement>} td
- * @param {enum} type
  */
-function onClick(td, type) {
+function onClick(td) {
   if ($(td).hasClass('na')) {
     return;
   }
   const colorFunctionDiv = $('#color-fxn-editor');
+  // open -> closed
   if (colorFunctionDiv.is(':visible') && td === globalTd) {
+    maybeDisplayWarningOnClose();
     colorFunctionDiv.hide();
     selectCurrentRow(false);
     return;
   }
-  colorFunctionDiv.show();
+  // most recent closed -> open
   if (td === globalTd) {
+    colorFunctionDiv.show();
     selectCurrentRow(true);
     return;
   }
+  // open td other than most recent closed
+  if (colorFunctionDiv.is(':visible')) {
+    maybeDisplayWarningOnClose();
+  }
+  colorFunctionDiv.show();
   selectCurrentRow(false);
   globalTd = td;
   selectCurrentRow(true);
+  const {currentStyle} = getColorFunction();
   $('#' + colorStyleTypeStrings.get(type) + '-radio').prop('checked', true);
```

Comment:
looks like type variable still needed here. I'm getting JavaScript errors loading page now.

Reply:
Sorry, forgot to rename that one. Thanks!